orchagent/vnetorch.cpp

```diff
@@ -1055,6 +1064,97 @@ bool VNetRouteOrch::doRouteTask<VNetVrfObject>(const string& vnet, IpPrefix& ipP
     return true;
 }
 
+void VNetRouteOrch::addPriorityEndpoints(const string& vnet, IpPrefix& ipPrefix, NextHopGroupKey& primary,  NextHopGroupKey& all)
+{
+    std::set<NextHopKey> nhks_primary = primary.getNextHops();
+    std::set<NextHopKey> nhks_all = all.getNextHops();
+    for (auto nkh : nhks_all)
+    {
+        bool is_primary = std::find(nhks_primary.begin(), nhks_primary.end(),nkh) != nhks_primary.end();
+        PriorityEpInfo ep_info;
+        ep_info.is_primary = is_primary;
+        ep_info.tunnel_prefix = ipPrefix;
+        priority_endpoints_[vnet][nkh.ip_address].push_back(ep_info);
+    }
+}
+
+void VNetRouteOrch::removePriorityEndpoints(const string& vnet, IpPrefix& ipPrefix)
+{
+    auto it_route = syncd_tunnel_routes_[vnet].find(ipPrefix);
+    if (it_route == syncd_tunnel_routes_[vnet].end())
+    {
+        SWSS_LOG_INFO("Failed to find priority tunnel route entry, prefix %s\n",
+        ipPrefix.to_string().c_str());
+        return;
+    }
+    auto old_primary_nhgk = it_route->second.primary;
+    auto old_all_nhgk = it_route->second.all;
+
+    std::set<NextHopKey> nhks_primary = old_primary_nhgk.getNextHops();
+    for (auto nhk: nhks_primary)
```

Comment:
Not sure I understand this logic. You want to erase `ipPrefix `from `priority_endpoints_`, right? Why not just loop and erase?

Reply:
yes, I could use the iterator to erase it. ill change it.

Reply:
done